extensions/amp-form/0.1/amp-form.js

```diff
@@ -72,13 +79,27 @@ export class AmpForm {
     }
     if (this.xhrAction_) {
       e.preventDefault();
+      this.setState_(FormState_.SUBMITTING);
       this.xhr_.fetchJson(this.xhrAction_, {
         body: new FormData(this.form_),
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      });
+      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
```

Comment:
I was going to do it in this PR but thought I'll follow this one up with that change.